### ISSUE TITLE
"My Health Log" symptoms checkboxes do not accurately convey the checked/unchecked state [Android]

### ISSUE THREAD

**Issue by jonnyjames** [covid-alert-ny/covid-green-app#245]
The checkboxes on the "My Health Log" screen (health Symptoms List) does not accurately convey the checked/unchecked state of the checkbox controls.

Currently, an unchecked checkbox is announced by TalkBack as "Muscles aches or body aches, unselected, muscle aches or body aches, unselected, checkbox" (the label and state are repeated twice). When a user double taps on the checkbox to change its state to checked/unchecked, TalkBack doesn't announce the change. If the user then moves focus away from the control, and back again, the now "checked" checkbox is still announced as "Muscles aches or body aches, unselected, muscle aches or body aches, unselected, checkbox".

![My Health Log - NRV - Checkbox unselected](https://user-images.githubusercontent.com/7034772/92038552-94016d00-ed41-11ea-9047-05107f701c34.png)

**Remediation Guidance:**

Use the `accessibilityState` attribute to convey the `checked` state to TalkBack users.

```javascript
<TouchableOpacity
  onPress={…}
  accessibilityLabel='Fatigue, Do you have any of the following 11 symptoms today?'
  accessibilityRole='checkbox'
  accessibilityState={{checked: true}}
>
  {/* Checkbox content */}
</TouchableOpacity>
```

**Comment by floridemai** [comment]
@jonnyjames This has now been addressed.
Note/question: When i double tap to check an item the VoiceOver says: "Muscles aches or body aches, checkbox, not checked" (telling what I am taping on) and then "Muscles aches or body aches, checkbox, checked" - the updated version. Is the correct?

**Comment by jonnyjames** [comment]
@floridemai The checkbox after it receives focus, and announces "Fatigue, do you have any of the following 11 symptoms today, checkbox, not checked", then if the TalkBack users double taps on the checkbox it announced "not checked, checked". It should only announce "checked".

**Comment by floridemai** [comment]
@jonnyjames I was wondering about two things:
1) Should we still rate this as a high priority in the state that it is?
2) We've looked into solving this and @AlanSl found https://www.npmjs.com/package/react-native-accessible-selectable which says:
> On double tap VoiceOver will read the content of the focused element. When you try to build a custom checkbox and make it accessible, you probably want to add selected to the accessibilityTraits, depending on whether the checkbox is currently checked or not. This works well on focus. VoiceOver will announce the state of your component. But on double tap it will first read the old state and then the new one. This can be quite confusing for users.
> The underlying problem is, that the decision to change the state from "not selected" to "selected" (or vice versa) is done in JavaScript and runs asynchronously after the tap. So VoiceOver will start to read before the state has changed.

Do you know a better and simpler way of achieving what we want in this case?

**Comment by jonnyjames** [comment]
@floridemai I want to clarify, are we referring to the Android or the iOS app? The content above is referencing "VoiceOver" (iOS), and not TalkBack.

**Comment by floridemai** [comment]
@jonnyjames My bad, iOS. Let me add it to the other ticket...

**Comment by floridemai** [comment]
@willslone Is this ok for iOS? Any guidance?

**Comment by jonnyjames** [comment]
@floridemai The correct state is announced, but as I mentioned previously:

> @floridemai The checkbox after it receives focus, and announces "Fatigue, do you have any of the following 11 symptoms today, checkbox, not checked", then if the TalkBack users double taps on the checkbox it announced "not checked, checked". It should only announce "checked".

Let me know if this will be resolved.

**Comment by AlanSl** [comment]
@jonnyjames We have a fix for this pending review, it didn't pass review in time for today's build but it should be in tomorrow's.

**Comment by AlanSl** [comment]
@jonnyjames Fix mentioned above is merged so this should be resolved in the next build.

**Comment by jonnyjames** [comment]
@AlanSl This issue is now fixed. Closing this issue.
